docs/src/pages/components/data-grid/pagination/PageInitialState.js

```diff
@@ -0,0 +1,27 @@
+import * as React from 'react';
+import { DataGrid } from '@mui/x-data-grid';
+import { getSynchronousDemoData } from '@mui/x-data-grid-generator';
+
+const data = getSynchronousDemoData({
```

Comment:
This function blocks the main thread. It can't be used to generate more than a couple thousands rows. It maybe only used in this very specific demo, which makes it overkill to add to `@mui/x-data-grid-generator`. You could use `useDemoData` and delay the rendering of the DataGrid once it has data: 

```tsx
const { data, loading } = useDemoData();

if (loading) {
  return null;
}
```

Or load the data from a JSON file. Check the `packages/grid/x-data-grid-generator/src/datagen-cli.ts` script (you need to fix some paths to make it  run).